### DISCUSSION TITLE
VS Code Codespaces - No access to GitHub Copilot found.

### DISCUSSION THREAD

**Discussion by bgolubovic** [community/community#41091] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Hi,

I have GitHub Copilot on my personal GitHub account. I am using my biz account to login to GitHub Codespaces with VS Code.
In VS Code I am logged in with booth accounts. I installed GitHub Copilot extension locally and in Codespaces.
But I am getting the error: "No access to GitHub Copilot found."

![Screenshot 2022-12-08 at 14 05 58](https://user-images.githubusercontent.com/12097483/206454196-25d58010-352b-4a67-863c-4fe9ce2267a7.png)
![Screenshot 2022-12-08 at 14 06 03](https://user-images.githubusercontent.com/12097483/206454201-eb298100-5613-4d33-abb7-f324fe065e6e.png)
![Screenshot 2022-12-08 at 14 06 12](https://user-images.githubusercontent.com/12097483/206454203-584faa15-2681-4ae9-a5f7-d91508df8256.png)

Setting Sync is on with my GitHub personal account where I have subscription for Copilot.

Any help?

Thank you.

**Comment by iamcjwilliams** [comment]
@bgolubovic Sorry for the trouble here, are able to use copilot but just logging into your account and not codespaces?

**Reply by bgolubovicsymphony** [reply]
I tried outside of codespace, in other regular VS Code projects, and I got a different error:
`GitHub Copilot could not connect to server. Extension activation failed: "Failed to get copilot token"`

**Reply by iamcjwilliams** [reply]
@bgolubovic Have you tried logging in and out of your account to reconnect to the server?

**Comment by bgolubovic** [comment]
@iamcjwilliams I did. And on a regular instance of VS Code, it is working fine. But in Codespaces (with devcontainers) not. I think it is trying to connect inside codespaces created under other GitHub accounts. 

Thank you for your time and answers.

**Reply by iamcjwilliams** [reply]
Are you able to login into codespaces with the same copilot account?

**Reply by bgolubovicsymphony** [reply]
Codespace account and copilot account are two different ones. I assume this is an issue? Does it need to be the same?

**Comment by sc-mitton** [comment]
For anyone else reading this thread, I got the same error, and my issue was that I needed to actually sign up for copilot in github first.

**Reply by bgolubovicsymphony** [reply]
How did you do that? Thank you.

**Reply by Jbalber** [reply]
That's the way I signed up, through GitHub for the 60-day trial. I use it in wsl2 Ubuntu. Or want to. When I open VS code I have to create a new wsl2 window. The first time installing, It synced everything to Git Hub, in the settings. And Copilot errors went way. When i opened Vs Code an hour or so again, all the errors came back and nothing changes after a sync. 'No access to Github found'

**Comment by chengweijian** [comment]
You should be logged into vscode and attached to github. it

**Comment by bgolubovic** [comment]
The latest testing and discovery is that uninstalling GitHub Copilot VS Code extension and installing the new GitHub Copilot Nightly it start working in regular VS code, without devcontainers and outside Codespaces.

Inside the Codespaces still the same issue. Because Codespaces access is with other GitHub account without GitHub Copilot.

**Reply by bgolubovic** [reply]
![Screenshot 2023-02-16 at 12 24 37](https://user-images.githubusercontent.com/12097483/219352468-9ba5016a-d87f-4816-a0de-5aae15469735.png)

**Comment by adityas1335** [comment]
If you don't yet have Copilot access, you can [[request it here](https://github.com/login?return_to=%2fgithub-copilot%2fsignup)]. If you are a student, you can get Copilot for FREE [following these instructions](https://techcommunity.microsoft.com/t5/educator-developer-blog/step-by-step-setting-up-github-student-and-github-copilot-as-an/ba-p/3736279?WT.mc_id=academic-0000-sagibbon).

To ensure that Copilot is running correctly, navigate to the extension tab in your Codespace and check the status of the Copilot extension. If the status is inactive, recreate the Codespace, and enable the extension to ensure that it is activated.
